core/sugartypes.ml

```diff
@@ -483,6 +484,22 @@ struct
     | Fun (bndr, _, (_, fn), _, _) ->
        let name = singleton (Binder.to_name bndr) in
        name, (diff (funlit fn) name)
+    | Mutual bnds ->
+        (* Traverse, get names and RHSes *)
+        let names, fnlits, hlits =
+          List.fold_right
+            (fun b (names, fnlits, handlerlits) ->
+              match WithPos.node b with
+                | (Fun (bndr, _, (_, rhs), _, _)) ->
+                   (add (Binder.to_name bndr) names, rhs::fnlits, handlerlits)
+                | (Handler (bndr, hl, _)) ->
```

Comment:
I want to get rid of the `Handler` node. It was a poor attempt at simulating first-class handlers using functions. I wouldn't be opposed to you just doing `assert false` here. Anyways, I will get rid of the `Handler` node in a subsequent patch.